torch/testing/_internal/common_utils.py

```diff
@@ -1542,21 +1381,66 @@ def maybe_to_list(a: Any) -> Any:
             x = maybe_to_tensor(x)
             y = maybe_to_tensor(y)
 
-            if isinstance(x, torch.Tensor) and isinstance(y, torch.Tensor):
-                self.assertEqual(
-                    x, y, atol=atol, rtol=rtol, msg=msg, exact_dtype=exact_dtype, exact_device=exact_device
-                )
+            if not (isinstance(x, torch.Tensor) and isinstance(y, torch.Tensor)):
+                # In case we can't convert the inputs to a tensor, we fall back to comparing x and y as iterables
+                x = maybe_to_list(x)
+                y = maybe_to_list(y)
+
+            self.assertEqual(
+                x,
+                y,
+                msg=msg,
+                atol=atol,
+                rtol=rtol,
+                equal_nan=equal_nan,
+                exact_dtype=exact_dtype,
+                exact_device=exact_device,
+                exact_stride=exact_stride,
+                exact_is_coalesced=exact_is_coalesced,
+            )
+        elif isinstance(x, (Number, bool, np.bool_)) or isinstance(y, (Number, bool, np.bool_)):
+            # torch.testing.assert_close does not allow Tensor vs. scalar comparisons
```

Comment:
Another option would be to verify this is the torch.Tensor x scalar or scalar x scalar case and then use torch.result_type to determine the comparison dtype